cluster_management/src/main/java/com/pinterest/rocksplicator/LeaderFollowerStateModelFactory.java

```diff
@@ -471,6 +478,46 @@ public void onBecomeFollowerFromOffline(Message message, NotificationContext con
           Utils.restoreLocalDBFromS3(adminPort, dbName, s3Bucket, s3Path, snapshotHost,
               snapshotPort);
         }
+
+        try {
+          LOG.error(dbName + " Catching up live updates to leader");
+          Map<String, String> liveHostAndRole = getLiveHostAndRole(context, dbName);
+          // Find upstream, prefer Leader
+          String upstream = null;
+          for (Map.Entry<String, String> instanceNameAndRole : liveHostAndRole.entrySet()) {
+            String role = instanceNameAndRole.getValue();
+            if (role.equalsIgnoreCase("LEADER")) {
+              upstream = instanceNameAndRole.getKey();
+              break;
+            } else {
+              upstream = instanceNameAndRole.getKey();
+            }
+          }
+
+          String upstreamHost = (upstream == null ? LOCAL_HOST_IP : upstream.split("_")[0]);
+          LOG.error(dbName + " got leader " + upstreamHost);
+          int upstreamPort =
+              (upstream == null ? adminPort : Integer.parseInt(upstream.split("_")[1]));
+
+          long local_seq_num;
+          long leader_seq_num;
+          // wait for up to 10 mins
+          for (int i = 0; i < 600; ++i) {
```

Comment:
We could add a stats here to indicate whether such scenario occurs or not. Otherwise, since no exception throw, we won't know whether this 10mis catch up or not till we grep the log.

Reply:
Yes, I can add stats here.